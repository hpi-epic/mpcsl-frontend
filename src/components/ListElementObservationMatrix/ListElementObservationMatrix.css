@import '~antd/dist/antd.css';

.Card {
	margin-left: 20px;
	border-radius: 25px;
}

.List-Buttons {
	float: right;
	margin-right: 5px;
}

.Card-Title {
	float: left;
}

.Card-Content {
	float: left;
}
