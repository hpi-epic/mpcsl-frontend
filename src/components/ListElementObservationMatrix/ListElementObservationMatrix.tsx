import { Button, Card, List } from 'antd';
import React from 'react';
import './ListElementObservationMatrix.css';

interface IPropsListElementObservationMatrix {
  title: string;
  content: string;
  onDelete: () => void;
  onView: () => void;
}

function ListElementObservationMatrix(
  props: IPropsListElementObservationMatrix,
) {
  const cardTitle = (
    <div>
      <h2 className='Card-Title'>{props.title}</h2>
    </div>
  );

  return (
    <div>
      <List.Item>
        <Card className='Card' title={cardTitle}>
          <p className='Card-Content'>{props.content}</p>
          <Button className='List-Buttons' onClick={props.onDelete} type='danger' ghost={true}>
            Delete
          </Button>
          <Button className='List-Buttons' onClick={props.onView} type='primary' ghost={true}>
            View
          </Button>
        </Card>
      </List.Item>
    </div>
  );
}

export default ListElementObservationMatrix;
