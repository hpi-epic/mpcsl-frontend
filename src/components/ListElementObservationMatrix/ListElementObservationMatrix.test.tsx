import React from 'react';
import renderer from 'react-test-renderer';
import ListElementObservationMatrix from './ListElementObservationMatrix';

describe('<ListElementObservationMatrix />', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <ListElementObservationMatrix
          title="Test ObservationMatrix"
          content="A Test ObeservationMatrix"
          onDelete={() => undefined}
          onView={() => undefined}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
