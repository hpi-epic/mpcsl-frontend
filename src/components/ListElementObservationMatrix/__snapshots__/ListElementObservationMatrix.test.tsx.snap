// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListElementObservationMatrix /> renders correctly 1`] = `
<div>
  <div
    className="ant-list-item"
  >
    <div
      className="ant-list-item-content ant-list-item-content-single"
    >
      <div
        className="ant-card Card ant-card-bordered"
      >
        <div
          className="ant-card-head"
          style={Object {}}
        >
          <div
            className="ant-card-head-wrapper"
          >
            <div
              className="ant-card-head-title"
            >
              <div>
                <h2
                  className="Card-Title"
                >
                  Test ObservationMatrix
                </h2>
              </div>
            </div>
          </div>
        </div>
        <div
          className="ant-card-body"
          style={Object {}}
        >
          <p
            className="Card-Content"
          >
            A Test ObeservationMatrix
          </p>
          <button
            className="ant-btn List-Buttons ant-btn-danger ant-btn-background-ghost"
            onClick={[Function]}
            type="button"
          >
            <span>
              Delete
            </span>
          </button>
          <button
            className="ant-btn List-Buttons ant-btn-primary ant-btn-background-ghost"
            onClick={[Function]}
            type="button"
          >
            <span>
              View
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
